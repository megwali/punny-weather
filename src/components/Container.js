import { useEffect, useState } from 'react';
import { Route, Switch, useLocation } from 'react-router-dom';
import { Locate, Details, Home, Search } from '../pages';
import { useSearchQuery, useUserLocation } from '../hooks';


const Container = () => {
  const [query, setQuery] = useState('');
  const { pathname } = useLocation();
  const [skip, setSkip] = useState(pathname !== '/locate');
  const locationForecast = useUserLocation(skip);
  const searchForecast = useSearchQuery(query);

  useEffect(() => {
    setSkip(pathname !== '/locate' || locationForecast?.summary?.length)
  }, [locationForecast?.summary, pathname])

  return (
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>

      <Route path="/locate">
        <Locate data={locationForecast} />
      </Route>

      <Route path="/search">
        <Search data={searchForecast} setQuery={setQuery} />
      </Route>

      <Route path="/:id">
        <Details />
      </Route>
    </Switch>
  );
};

export default Container;
