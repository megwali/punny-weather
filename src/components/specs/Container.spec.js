import { create } from 'react-test-renderer';
import Container from '../Container';

jest.mock('react-router-dom', () => ({ Route: 'Route', Switch: 'Switch', useLocation: () => ({ pathname: '/' }) }));
jest.mock('../../pages', () => ({ Dashboard: 'Dashboard', Details: 'Details', Home: 'Home', Search: 'Search' }));


describe('Container', () => {
  it('renders', () => {
    const component = create(<Container />);
    expect(component.toJSON()).toMatchSnapshot();
  })
});
