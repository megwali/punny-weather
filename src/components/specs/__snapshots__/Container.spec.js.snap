// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  >
    <Home />
  </Route>
  <Route
    path="/locate"
  >
    <Locate
      data={
        Object {
          "city": null,
          "error": false,
          "list": Array [],
          "loading": true,
          "summary": Array [],
        }
      }
    />
  </Route>
  <Route
    path="/search"
  >
    <Search
      data={
        Object {
          "city": null,
          "error": false,
          "list": Array [],
          "loading": false,
          "summary": Array [],
        }
      }
      query=""
      setQuery={[Function]}
    />
  </Route>
  <Route
    path="/:id"
  >
    <Details />
  </Route>
</Switch>
`;
